chart/templates/patch_serviceaccount.yaml

```diff
@@ -0,0 +1,64 @@
+{{- include "_config.load" $ }}
+{{- if gt (len .Values.kube.image_pull_secrets) 0 }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded
+  name: default
+  namespace: {{ .Release.Namespace | quote }}
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - serviceaccounts
+  verbs:
+  - get
+  - patch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded
+  name: default
+  namespace: {{ .Release.Namespace | quote }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: default
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: {{ .Release.Namespace | quote }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: patch-sa
+  namespace: {{ .Release.Namespace | quote }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: sa
+        image: google/cloud-sdk
+        imagePullPolicy: IfNotPresent
+# set image_pull_secrets list
+{{- $image_pull_secret_names := "" }}
+{{- range $secret_name := $.Values.kube.image_pull_secrets }}
+  {{- $image_pull_secret_names = cat $image_pull_secret_names (printf "{\"name\":%s}," ($secret_name | quote)) }}
+{{- end }}
+        command: ["/bin/sh", "-c"]
+        args:
+        - |
+          kubectl patch serviceaccount default -p '{"imagePullSecrets": [{{ $image_pull_secret_names | trimSuffix "," }} ] }' -n {{ .Release.Namespace }}
+{{- end }}
```

Comment:
> I also added `| uniq` ...

Please ignore; I was confused and thought the list of secrets was assembled from different sources. But since they all need to be specified during deployment, the schema already guarantees that the elements are unique.